lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +211,42 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Return Doctrine/Mysql-compatible column default values for MariaDB 10.2.7+ servers.
+     *
+     * - Since MariaDb 10.2.7 column defaults stored in information_schema are now quoted
+     *   to distinguish them from expressions (see MDEV-10134 for a what is an expression).
+     * - CURRENT_TIMESTAMP, CURRENT_TIME, CURRENT_DATE are stored in information_schema
+     *   as current_timestamp(), currdate(), currtime()
+     * - Literal escaping is normalized in information schema (store "''" instead of "\'")
+     *
+     * @link https://mariadb.com/kb/en/library/information-schema-columns-table/
+     * @link https://jira.mariadb.org/browse/MDEV-13132
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(MariaDb102Platform $platform, ?string $columnDefault) : ?string {
+        if ($columnDefault === 'NULL' || $columnDefault === null) {
```

Comment:
Could you please use early-returns here and simplify this method?

Reply:
Something like this ?
```php
        if ($columnDefault === 'NULL' || $columnDefault === null) {
            return null;
        }
        if (strpos($columnDefault, "'") === 0) {
            return stripslashes(
                str_replace("''", "'",
                    preg_replace('/^\'(.*)\'$/', '$1', $columnDefault)
                )
            );
        }
        if ($columnDefault === 'current_timestamp()') {
            return $platform->getCurrentTimestampSQL();
        }
        if ($columnDefault === 'curdate()') {
            return $platform->getCurrentDateSQL();
        }
        if ($columnDefault === 'curtime()') {
            return $platform->getCurrentTimeSQL();
        } 
        return $columnDefault;
```

Reply:
Do you have a suggestion for the `curr_()` tests ? A switch ? An array with values ?

Reply:
Like this ?

```php
        switch($columnDefault) {
            case 'current_timestamp()';
                return $platform->getCurrentTimestampSQL();
            case 'curdate()':
                return $platform->getCurrentDateSQL();
            case 'curtime()':
                return $platform->getCurrentTimeSQL();
        }

```